rocprim/include/rocprim/device/detail/device_segmented_radix_sort.hpp

```diff
@@ -39,6 +39,14 @@
 
 BEGIN_ROCPRIM_NAMESPACE
 
+// Currently HIP on Windows has a bug involving inline device functions generating
+// local memory/register allocation errors during compilation.  Current workaround is to
+// use __attribute__((always_inline)) for the affected functions
+#ifdef WIN32
+#define ROCPRIM_SEGMENTED_RADIX_SORT_INLINE inline __attribute__((always_inline))
```

Comment:
Why only for segmented radix sort? Should we use a global one?

Reply:
At the moment, I only see the compilation error for segmented radix sort, so I decided to keep the workaround as minimal as possible to avoid any side effects of making all device functions __attribute__((always_inline)).  But that is with my current windows test machine, HIP on Windows CI hasn't been set up for rocPRIM yet so its quite possible I have to add the attribute to more or all of the rocPRIM device functions...

Reply:
Background on this is at https://github.com/ROCmSoftwarePlatform/rocBLAS/issues/1193 but the timeline for any fix to get into the llvm/clang we use is not known.   Additional functions may fail in the near future ( kernels that call other device functions ) or may not, it is not easy to predict.    Longer term these attributes should go away.

Reply:
Ok thanks.